### DISCUSSION TITLE
support for Pages Router (Next version 13)

### DISCUSSION THREAD

**Discussion by winatungmiharja** [47ng/nuqs#636] [Q&A] [answered]
```
"next": "13.4.19"
"nuqs": "^1.19.1",
```
I'm attempting to utilize the `nuqs` library in my Next.js project (pages router). However, the demo code provided in the `nuqs` documentation (which is supposed for app routers), is not functioning as expected. The component keep unmounting which results in the input not being focused again (see video below)

```tsx
import * as React from 'react';
import { useQueryState } from 'nuqs';

export default function Demo() {
  const [hello, setHello] = useQueryState('hello', { defaultValue: '', shallow: true });

  return (
    <div className='layout py-20'>
      <pre>{JSON.stringify(hello, null, 2)}</pre>
      <input
        autoComplete='off'
        value={hello}
        placeholder='Enter your name'
        onChange={(e) => setHello(e.target.value)}
        className='w-full rounded border border-gray-300 p-2'
      />
    </div>
  );
}

```
As I type, the input re-mount and keep unfocused, so I had to focus again and type. After 1-2 character, it happens again

https://github.com/user-attachments/assets/b3de99fa-f976-404f-9e7e-b8f105ae75cc

I think `nuqs` support pages router as they stated it in the readme
![image](https://github.com/user-attachments/assets/fbc93d5c-5714-4464-a2bb-84fae4db66a8)

# Question
Am I missing some part about pages router in the documentation? Or am I making some technical mistakes in my syntax?

Help is really appreciated🙏🏻

**Comment by franky47** [comment, marked as answer]
I did a clean install of `next@13.4.19` and tried your example, but it works fine on my end. Could there be something else causing the component to remount?

https://github.com/user-attachments/assets/e78d8ce4-a0a3-4de4-b02d-477be5083c1d

**Reply by winatungmiharja** [reply]
I think there must be something from my next.js config that causing this issue. I'll figure it out!

Thankyou for your help!
